src/client/options/mod.rs

```diff
@@ -871,11 +870,24 @@ impl TlsOptions {
                 }
             };
 
+            ClientConfig::builder()
```

Comment:
`ClientConfig` was made mostly opaque in favor of a builder, documentation [here](https://docs.rs/rustls/latest/rustls/struct.ConfigBuilder.html). This branch calls `with_single_cert` on the builder which corresponds to the `set_single_client_cert` method we were calling on the config previously. The `else` branch uses the defaults suggested in the docs, which are also in line with the now-removed [`new`](https://docs.rs/rustls/0.18.0/rustls/struct.ClientConfig.html#method.new) function we were calling previously.